lucid/modelzoo/vision_base.py

```diff
@@ -150,10 +144,16 @@ def synsets(self):
 
   @property
   def name(self):
-    return self.__class__.name
+    if self.modelname == None:
+      return self.__class__.__name__
+    else:
+      return self.modelname
 
   def __str__(self):
-    return self.__class__.name
+    if self.modelname == None:
```

Comment:
you should change this block to just be `self.name`

Reply:
Done